lsp/src/service/types.ts

```diff
@@ -0,0 +1,18 @@
+import {
+    CompletionItem,
+    CompletionList,
+    Diagnostic,
+    Hover,
+    HoverParams,
+    TextDocumentPositionParams
+} from 'vscode-languageserver'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+
+export interface LanguageService {
```

Comment:
Consider creating an abstract class that service implementations can extend -- imagine that six months from now we add a 4th service event, we now have to back-fill a no-op into the existing services.